### PR TITLE
fix(azure-list-images): skip manually created images

### PR DESCRIPTION
**Pull request by soyacz**
When scylla image is built manually, build_id gets date
as build_id (could be also something else). In that case
we don't want to use this image in SCT when specifying
Scylla version.

Fix by skipping images when we cannot parse build_id as int.

fixes issue from comment: https://github.com/scylladb/scylla-cluster-tests/pull/5151#discussion_r951042994

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/provision/azure/utils.py [hunk 1]
```diff
@@ -71,7 +62,11 @@ def get_scylla_images(  # pylint: disable=too-many-branches,too-many-locals
         for image in gallery_image_versions:
             if image.location != region_name or image.name.startswith('debug-'):
                 continue
-
+            try:
+                int(image.tags.get("build_id"))
+            except (ValueError, TypeError):
+                # skip images with invalid build_id (e.g. created manually)
```
Comment (vponomaryov) [inline sdcm/provision/azure/utils.py:68]: Please, add "debug" logging.

Comment (soyacz) [reply sdcm/provision/azure/utils.py:68]: There's around 170 images from 4.6/4.7/others from year 2021 where Azure was not supported. Do we want these to pollute logs?

Comment (fruch) [reply sdcm/provision/azure/utils.py:68]: No, there's no point in that.

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**